packages/node/src/exec/win32.ts

```
@@ -1,7 +1,119 @@
+    detached: false,
+    // When windows shell is true, it runs with cmd.exe by default, but
+    // node has trouble with PATHEXT and exe. It can't run exe directly for example.
+    // `cross-spawn` handles running it with the shell in windows if needed.
+    // Neither mac nor linux need shell and we run it detached.
+    shell: false,
+    // With stdio as pipe, windows gets stuck where neither the child nor the
+    // parent wants to close the stream, so we call it ourselves in the exit event.
+    stdio: "pipe",
+    env: options.env,
+    cwd: options.cwd,
+  });
+
+  let { pid } = childProcess;
+
+  return yield resource(
+    { pid, stdin, stdout, stderr, join, expect },
+    function*() {
+      let onError = (value: unknown) =>
+        getResult.resolve({ type: "error", value });
+
+      try {
+        childProcess.on("error", onError);
+
+        yield spawn(
+          on<[string]>(childProcess.stdout, "data").forEach(function*([data]) {
+            addToTail(data);
+            stdout.send(data);
+          })
+        );
+
+        yield spawn(
+          on<[string]>(childProcess.stderr, "data").forEach(function*([data]) {
+            addToTail(data);
+            stderr.send(data);
+          })
+        );
+
+        yield spawn(
+          subscribe(stdin).forEach(function*(data) {
+            childProcess.stdin.write(data);
+          })
+        );
+
+        let value = yield once(childProcess, "exit");
+        getResult.resolve({ type: "status", value });
+      } finally {
+        stdout.close();
+        stderr.close();
+        childProcess.off("error", onError);
+        try {
+          // @ts-ignore
+          childProcess.stdout.end();
+          // @ts-ignore
+          childProcess.stderr.end();
+          // just in case, super kill the whole tree
+          // this isn't graceful but *shrug*
+          childProcess.kill('SIGTERM');
+          exec("taskkill /pid " + childProcess.pid + " /T /F");
```

If this rejects, it will cause an `UnhandledPromiseRejectionError`, so I think you'll need to just swallow the exception with a catch()